src/Estimators/include/BipedalLocomotion/RobotDynamicsEstimator/SubModelKinDynWrapper.h

```diff
@@ -82,25 +84,31 @@ class SubModelKinDynWrapper
     bool updateDynamicsVariableState();
 
 public:
+    /**
+     * @brief set kinDyn
+     * @param kinDyn is an iDynTree KinDynComputation object handling the kinematics and the
+     * dynamics of the model.
+     * @return a boolean value saying if the input pointer is valid
+     */
+    bool setKinDyn(std::shared_ptr<iDynTree::KinDynComputations> kinDyn);
+
     /**
      * @brief initialize initializes the kindyncomputation object and resizes
      * all the variable not allocated yet.
      * @param subModel a struct of type SubModif el describing the subModel structure and model.
-     * @param kinDynFullModel is the KinDynComputation object associated to a model.
      * @return a boolean value saying if all the variables are initialized correctly.
      */
     bool
-    initialize(SubModel& subModel, std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+    initialize(const SubModel& subModel);
 
     /**
      * @brief updateKinDynState updates the state of the KinDynWrapper object.
-     * @param kinDynFullModel is the KinDynComputation object associated to a model.
      * @return a boolean value saying if the subModelList has been created correctly.
      */
-    bool updateKinDynState(std::shared_ptr<iDynTree::KinDynComputations> kinDynFullModel);
+    bool updateKinDynState();
```

Comment:
To be honest, it is not clear to me what you are supposed to do in order to update the internal state. Will this class update the shared object, or it expects someone else on the outside to update the ``KinDynComputations`` object?

Reply:
It expects someone else to update the shared `KinDynComputations` object from outside since it is related to the full model. The logic I used is similar to what was done in https://github.com/ami-iit/bipedal-locomotion-framework/blob/5251bd9ac317c8617423385af97e06f2b3886742/src/IK/include/BipedalLocomotion/IK/IKLinearTask.h#L40.

The `KinDynComputation` object is set and someone else updates its state from outside before calling the methods using it inside.

Reply:
I go for `updateInternalKinDynState`. Thanks.